socket/EzySocketWriter.cs

```diff
@@ -11,18 +11,53 @@ protected override void update()
             while(true)
             {
                 if (!active)
+                {
                     return;
+                }
                 EzyPacket packet = packetQueue.take();
                 if (packet == null)
+                {
                     return;
+                }
                 int writtenBytes = writeToSocket(packet);
                 
                 packet.release();
                 if (writtenBytes <= 0)
+                {
                     return;
+                }
+                addSocketWriteStats(writtenBytes);
+            }
+        }
 
-                networkStatistics.getSocketStats().getNetworkStats().addWrittenPackets(1);
-                networkStatistics.getSocketStats().getNetworkStats().addWrittenBytes(writtenBytes);
+        public override bool call()
+        {
+            try
+            {
+                if (!active)
+                {
+                    return false;
+                }
+                EzyPacket packet = packetQueue.poll();
+                if (packet == null)
+                {
+                    return true;
+                }
+                int writtenBytes = writeToSocket(packet);
```

Comment:
Could you explain why we don't need `writeToSocketAsync` but we need `readSocketDataAsync`?

Reply:
Because `write` is not blocking or only blocking very short time so we don't need to use `write async`